helpers/pedigree_from_sample_metadata.py

```diff
@@ -0,0 +1,215 @@
+"""
+generate a ped file on the fly using the sample-metadata api client
+optional argument will remove all family associations
+    - family structure removal enforces singleton structure during this MVP
+optional argument will replace missing parents with '0' to be valid PLINK structure
+"""
+import json
+from typing import Dict, List, Union
+from itertools import product
+import logging
+import click
+from sample_metadata.apis import FamilyApi, ParticipantApi
+
+
+# the keys provided by the SM API, in the order to write in output
+PED_KEYS = [
+    'family_id',
+    'individual_id',
+    'paternal_id',
+    'maternal_id',
+    'sex',
+    'affected',
+]
+
+
+def get_fat_pedigree(
+    pedigree_dicts: List[Dict[str, Union[str, List[str]]]],
+    sample_to_cpg_dict: Dict[str, List[str]],
+    singles: bool,
+    plink: bool,
+) -> List[Dict[str, List[str]]]:
+    """
+    swaps sample IDs for the internal CPG values
+    optionally make all members unrelated singletons
+    optionally substitute missing member IDs for '0'
+
+    :param pedigree_dicts:
+    :param sample_to_cpg_dict:
+    :param singles:
+    :param plink:
+    :return:
+    """
+
+    new_entries = []
+
+    for counter, ped_entry in enumerate(pedigree_dicts, 1):
+
+        if ped_entry['individual_id'] not in sample_to_cpg_dict:
+            continue
```

Comment:
This is an error though right - or maybe the participant doesn't have any samples. I think a comment here might explain this case.
You might still want to proceed through, to show two people are related through a dummy individual

Reply:
This is the main actionable one for me :)

Reply:
replaced this by catching all failures, and throwing an exception at the end of the iteration to say 'these samples came from one endpoint, but not the other'.